code/game/turfs/turf.dm

```diff
@@ -644,3 +644,14 @@ GLOBAL_LIST_EMPTY(station_turfs)
 		if(turf_to_check.density || LinkBlockedWithAccess(turf_to_check, caller, ID))
 			continue
 		. += turf_to_check
+
+/turf/proc/GetHeatCapacity()
+	. = heat_capacity
+
+/turf/proc/GetTemperature()
+	. = temperature
+
+/turf/proc/TakeTemperature(temp)
+	temperature += temp
+	if(!blocks_air)
+		air_update_turf(FALSE, FALSE)
```

Comment:
a closed turf should never be added to active, and an open turf already is. don't do this